src/uct/api/v2/uct_v2.h

```diff
@@ -1083,6 +1083,182 @@ ucs_status_t
+    UCT_IFACE_ATTR_FIELD_TAG_EAGER          = UCS_BIT(4),
+
+    /** Enables @ref uct_iface_attr_v2_t::tag::rndv */
+    UCT_IFACE_ATTR_FIELD_TAG_RNDV           = UCS_BIT(5),
+
+    /** Enables @ref uct_iface_attr_v2_t::atomic32::op_flags */
+    UCT_IFACE_ATTR_FIELD_ATOMIC32_OP_FLAGS  = UCS_BIT(6),
+
+    /** Enables @ref uct_iface_attr_v2_t::atomic32::fop_flags */
+    UCT_IFACE_ATTR_FIELD_ATOMIC32_FOP_FLAGS = UCS_BIT(7),
+
+    /** Enables @ref uct_iface_attr_v2_t::atomic64::op_flags */
+    UCT_IFACE_ATTR_FIELD_ATOMIC64_OP_FLAGS  = UCS_BIT(8),
+
+    /** Enables @ref uct_iface_attr_v2_t::atomic64::fop_flags */
+    UCT_IFACE_ATTR_FIELD_ATOMIC64_FOP_FLAGS = UCS_BIT(9),
+
+    /** Enables @ref uct_iface_attr_v2_t::flags*/
+    UCT_IFACE_ATTR_FIELD_FLAGS              = UCS_BIT(10),
+
+    /** Enables @ref uct_iface_attr_v2_t::event_flags*/
+    UCT_IFACE_ATTR_FIELD_EVENT_FLAGS        = UCS_BIT(11),
+
+    /** Enables @ref uct_iface_attr_v2_t::device_addr_len */
+    UCT_IFACE_ATTR_FIELD_DEVICE_ADDR_LEN    = UCS_BIT(12),
+
+    /** Enables @ref uct_iface_attr_v2_t::iface_addr_len */
+    UCT_IFACE_ATTR_FIELD_IFACE_ADDR_LEN     = UCS_BIT(13),
+
+    /** Enables @ref uct_iface_attr_v2_t::ep_addr_len*/
+    UCT_IFACE_ATTR_FIELD_EP_ADDR_LEN        = UCS_BIT(14),
+
+    /** Enables @ref uct_iface_attr_v2_t::max_num_eps */
+    UCT_IFACE_ATTR_FIELD_MAX_NUM_EPS        = UCS_BIT(15),
+} uct_iface_attr_field_t;
+
+/**
+ * @ingroup UCT_RESOURCE
+ * @brief Interface attributes: capabilities and limitations.
+ */
+typedef struct uct_iface_attr_v2 {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref uct_iface_attr_field_t.
+     */
+    uint64_t            field_mask;
+
+    /** Attributes for PUT operations */
+    uct_iface_op_attr_t *put;
+
+    /** Attributes for GET operations */
+    uct_iface_op_attr_t *get;
+
+    /** Attributes for AM operations */
+    uct_iface_op_attr_t *am;
+
+    /** Attributes for TAG operations */
+    struct {
+        /** Attributes related to receive protocol */
+        uct_iface_op_attr_t *recv;
+
+        /** Attributes related to eager protocol */
+        uct_iface_op_attr_t *eager;
+
+        /** Attributes related to rendezvous protocol */
+        uct_iface_op_attr_t *rndv;
+    } tag;
+
+    /** Attributes for atomic operations */
+    struct {
+        /** Attributes for atomic-post operations */
+        uint64_t op_flags;
+        /** Attributes for atomic-fetch operations */
+        uint64_t fop_flags;
+    } atomic32, atomic64;
+
+    /** Flags from @ref UCT_RESOURCE_IFACE_CAP */
+    uint64_t flags;
+
+    /** Flags from @ref UCT_RESOURCE_IFACE_EVENT_CAP */
+    uint64_t event_flags;
+
+    /** Size of device address */
+    size_t   device_addr_len;
+
+    /** Size of interface address */
+    size_t   iface_addr_len;
+
+    /** Size of endpoint address */
+    size_t   ep_addr_len;
+
+    /** Maximum number of endpoints */
+    size_t   max_num_eps;
```

Comment:
IMO would be better to move these fields right after field_mask and before the put, get,.. ops
and order the flags accordingly